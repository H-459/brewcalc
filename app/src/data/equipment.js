export const equipment = {
  name: "Pot (13 Gal/50 L) - BIAB",
  equipment_items: {
    brew_kettle: {
      name: "Pot (13 Gal/50 L) - BIAB",
      form: "Brew Kettle",
      maximum_volume: {
        value: 50,
        unit: "l",
      },
      loss: {
        value: 3.8232879,
        unit: "l",
      },
      boil_rate_per_hour: {
        value: 5.364,
        unit: "l",
      },
    },
  },

  /*boilSize: 36.019978,
  batchSize: 23.0154363,
  tunVolume: 50.0055779,
  tunWeight: 6.9989246,
  tunSpecificHeat: 0.12,
  topUpWater: "0.0000000",
  trubChillerLoss: 3.8232879,
  evapRate: 0.14923151,
  boilTime: "90.0000000",
  calcBoilVolume: "TRUE",
  lauterDeadspace: 0,
  topUpKettle: 0,
  hopUtilization: "100.0000000",
  coolingLossPct: 0.04,
  efficiency: 0.68,
  BIAB: false,


  tunWeight: "144.0000000",
  tunSpecificHeat: "0.3000000",
  displayTunTemp: "22.2 C",
  displayTunWeight: "4.08 kg",

  */
};
