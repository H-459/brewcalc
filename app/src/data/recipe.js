export const recipe = {
  name: "Aussie Ale",
  version: "1",
  type: "All Grain",
  brewer: "Steve Nicholls",
  asstBrewer: "",
  batch_size: {
    value: 23.02,
    unit: "l"
  },
  boil: {
    pre_boil_size: {
      value: 37.1203164,
      unit: "l"
    },
    boil_time: { value: 90, unit: "min" }
  },
  efficiency: { brewhouse: { value: 68, unit: "%" } },
  ingredients: {
    fermentable_additions: [
      {
        name: "Pale Malt (2 Row) UK",
        type: "grain",
        amount: { value: 4.4969891, unit: "kg" },
        yield: { potential: { value: 1.029992, unit: "sg" } },
        color: { value: 4.925, unit: "EBC" },
        origin: "United Kingdom",
        supplier: "",
        notes:
          "Base malt for all English beer styles\r\nLower diastatic power than American 2 Row Pale Malt",
        coarseFineDiff: "1.5000000",
        moisture: "4.0000000",
        diastaticPower: "45.0000000",
        protein: "10.1000000",
        maxInBatch: "100.0000000",
        recommendMash: false,
        ibuGalPerLb: "0.0000000",
        displayAmount: "4.50 kg",
        inventory: "3.90 kg",
        displayColor: "4.9 EBC",
        extractSubstitute: ""
      },
      {
        name: "Munich Malt - 20L",
        type: "grain",
        amount: { value: 518.8834, unit: "g" },
        yield: { potential: { value: 1.0345, unit: "sg" } },
        color: { value: 39.4, unit: "EBC" },
        origin: "US",
        supplier: "",
        notes:
          "Malty-sweet flavor characteristic and adds a orange to deep orange color to the beer.\r\nDoes not contribute signficantly to body or head retention.\r\nUse for: Bock, Porter, Marzen, Oktoberfest beers",
        coarseFineDiff: "2.8000000",
        moisture: "5.0000000",
        diastaticPower: "25.0000000",
        protein: "13.5000000",
        maxInBatch: "80.0000000",
        recommendMash: true,
        ibuGalPerLb: "0.0000000",
        displayAmount: "0.52 kg",
        inventory: "0.45 kg",
        displayColor: "39.4 EBC",
        extractSubstitute: "Amber Liquid Extract"
      },
      {
        name: "Caramel/Crystal Malt - 20L",
        type: "grain",
        amount: { value: 0.201788, unit: "kg" },
        yield: { potential: { value: 1.0345, unit: "sg" } },
        color: { value: 39.4, unit: "EBC" },
        origin: "US",
        supplier: "",
        notes:
          "Adds body, color and improves head retention.\r\nAlso called &#34;Crystal&#34; malt.",
        coarseFineDiff: "1.5000000",
        moisture: "4.0000000",
        diastaticPower: "0.0000000",
        protein: "13.2000000",
        maxInBatch: "20.0000000",
        recommendMash: false,
        ibuGalPerLb: "0.0000000",
        displayAmount: "0.20 kg",
        inventory: "0.17 kg",
        displayColor: "39.4 EBC",
        extractSubstitute: ""
      },
      {
        name: "Roasted Barley",
        type: "grain",
        amount: { value: 0.046123, unit: "kg" },
        yield: { potential: { value: 1.0253, unit: "sg" } },
        color: { value: 591.0, unit: "EBC" },
        addAfterBoil: false,
        origin: "US",
        supplier: "",
        notes:
          "Roasted at high temperature to create a burnt, grainy, coffee like flavor.\r\nImparts a red to deep brown color to beer, and very strong roasted flavor.\r\nUse 2-4% in Brown ales to add a nutty flavor, or 3-10% in Porters and Stouts for coffee flavor.",
        coarseFineDiff: "1.5000000",
        moisture: "5.0000000",
        diastaticPower: "0.0000000",
        protein: "13.2000000",
        maxInBatch: "10.0000000",
        recommendMash: false,
        ibuGalPerLb: "0.0000000",
        displayAmount: "0.05 kg",
        inventory: "0.04 kg",
        displayColor: "591.0 EBC",
        extractSubstitute: ""
      },
      {
        name: "Cane (Beet) Sugar",
        type: "sugar",
        amount: { value: 0.2613635, unit: "kg" },
        yield: { potential: { value: 1.046, unit: "sg" } },
        color: { value: 0, unit: "EBC" },
        origin: "US",
        supplier: "",
        notes:
          "Common household baking sugar.\r\nLightens flavor and body of beer.\r\nCan contribute a cider-like flavor to the beer if not cold-fermented or used in large quantities.",
        coarseFineDiff: "1.5000000",
        moisture: "4.0000000",
        diastaticPower: "120.0000000",
        protein: "11.7000000",
        maxInBatch: "7.0000000",
        recommendMash: false,
        ibuGalPerLb: "0.0000000",
        displayAmount: "0.26 kg",
        inventory: "0.25 kg",
        displayColor: "0.0 EBC",
        extractSubstitute: ""
      }
    ],
    hop_additions: [
      {
        name: "Pride of Ringwood",
        origin: "Australia",
        alpha_acid: {
          value: 10,
          unit: "%"
        },
        amount: { value: 5.2055, unit: "g" },
        timing: {
          // TODO: first wort???
          use: "add_to_boil",
          time: { value: 60, unit: "min" }
        },
        notes:
          "Use for: General purpose bittering hops for Australian beers\r\nAroma: Moderate citric aroma, clean bittering flavor\r\nSubstitutes: Cluster, Galena",
        type: "Both",
        form: "Leaf",
        beta: "5.8000000",
        hsi: "45.0000000",
        displayAmount: "5.21 g",
        inventory: "5.00 g",
        displayTime: "60.0 min"
      },
      {
        name: "Pride of Ringwood",
        origin: "Australia",
        alpha_acid: {
          value: 10,
          unit: "%"
        },
        amount: { value: 5.2055, unit: "g" },
        timing: {
          use: "add_to_boil",
          time: { value: 45, unit: "min" }
        },
        notes:
          "Use for: General purpose bittering hops for Australian beers\r\nAroma: Moderate citric aroma, clean bittering flavor\r\nSubstitutes: Cluster, Galena",
        type: "Bittering",
        form: "Pellet",
        beta: "5.8000000",
        hsi: "45.0000000",
        displayAmount: "5.21 g",
        inventory: "5.00 g",
        displayTime: "45.0 min"
      },
      {
        name: "Pride of Ringwood",
        origin: "Australia",
        alpha_acid: {
          value: 10,
          unit: "%"
        },
        amount: { value: 31.2328, unit: "g" },
        timing: {
          use: "add_to_boil",
          time: { value: 15, unit: "min" }
        },
        notes:
          "Use for: General purpose bittering hops for Australian beers\r\nAroma: Moderate citric aroma, clean bittering flavor\r\nSubstitutes: Cluster, Galena",
        type: "Both",
        form: "Leaf",
        beta: "5.8000000",
        hsi: "45.0000000",
        displayAmount: "31.23 g",
        inventory: "30.00 g",
        displayTime: "15.0 min"
      },
      {
        name: "Pride of Ringwood",
        origin: "Australia",
        alpha_acid: {
          value: 10,
          unit: "%"
        },
        amount: { value: 15.6818, unit: "g" },
        timing: {
          use: "add_to_boil",
          time: { value: 0, unit: "min" }
        },
        notes:
          "Use for: General purpose bittering hops for Australian beers\r\nAroma: Moderate citric aroma, clean bittering flavor\r\nSubstitutes: Cluster, Galena",
        type: "Bittering",
        form: "Leaf",
        beta: "5.8000000",
        hsi: "45.0000000",
        displayAmount: "15.68 g",
        inventory: "15.00 g",
        displayTime: "0.0 min"
      }
    ],
    culture_additions: [
      {
        name: "American Ale",
        type: "ale",
        form: "liquid",
        amount: 0.125048,
        amountIsWeight: false,
        laboratory: "Wyeast Labs",
        productId: "1056",
        minTemperature: "15.5555556",
        maxTemperature: "22.2222222",
        flocculation: "Medium",
        attenuation: { value: 75, unit: "%" },
        notes:
          "Soft, smooth, clean finish.  Very well balanced.  Very versitile -- works well with many ale styles.",
        bestFor:
          "American Pale Ale, Scottish Ale, Porters, Sweet Stout, Barley Wine, Alt",
        maxReuse: "5",
        timesCultured: "0",
        addToSecondary: false,
        displayAmount: "125.05 ml",
        dispMinTemp: "15.6 C",
        dispMaxTemp: "22.2 C",
        inventory: "0.0 pkg",
        cultureDate: "18 Jun 2003"
      }
    ]
  },
  style: {
    name: "Australian Ale",
    category: "Ale",
    categoryNumber: "1",
    styleLetter: "A",
    styleGuide: "",
    type: "Ale",
    ogMin: "1.0350000",
    ogMax: "1.0600000",
    fgMin: "1.0080000",
    fgMax: "1.0150000",
    ibuMin: "10.0000000",
    ibuMax: "30.0000000",
    colorMin: "3.9400000",
    colorMax: "19.7000000",
    carbMin: "2.2000000",
    carbMax: "2.8000000",
    abvMax: "5.0000000",
    abvMin: "2.0000000",
    notes: "Medium malt with slight grain dryness",
    profile: "",
    ingredients: "Pride of Ringwood hops.",
    examples: "Cooper&#39;s Pale Ale",
    displayOgMin: "1.035 SG",
    displayOgMax: "1.060 SG",
    displayFgMin: "1.008 SG",
    displayFgMax: "1.015 SG",
    displayColorMin: "3.9 EBC",
    displayColorMax: "19.7 EBC",
    ogRange: "1.035-1.060 SG",
    fgRange: "1.008-1.015 SG",
    ibuRange: "10.0-30.0 IBUs",
    carbRange: "2.20-2.80 Vols",
    colorRange: "3.9-19.7 EBC",
    abvRange: "2.00-5.00 %"
  },

  mash: {
    name: "Single Infusion, Medium Body",
    grain_temperature: {
      value: 22.22,
      unit: "C"
    },
    spargeTemp: 75.5555556,
    ph: "5.4000000",
    tunWeight: "144.0000000",
    tunSpecificHeat: "0.3000000",
    equipAdjust: true,
    notes:
      "Simple single infusion mash for use with most modern well modified grains (about 95% of the time).",
    displayGrainTemp: "22.2 C",
    displayTunTemp: "22.2 C",
    displaySpargeTemp: "75.6 C",
    displayTunWeight: "4.08 kg",
    mash_steps: [
      {
        name: "Mash In",
        type: "infusion",
        infuse_temperature: {
          value: 74.1,
          unit: "C"
        },
        amount: {
          value: 16.76,
          //value: 13.7276426,
          unit: "l"
        },
        step_time: {
          value: 60,
          unit: "min"
        },
        step_temperature: {
          value: 66.67,
          unit: "C"
        }
      },
      {
        name: "Mash Out",
        type: "infusion",
        infuse_temperature: {
          value: 98.5,
          unit: "C"
        },
        amount: {
          value: 7.6874799,
          unit: "l"
        },
        step_time: {
          value: 10,
          unit: "min"
        },
        step_temperature: {
          value: 75.55,
          unit: "C"
        }
      }
    ]
  },
  notes:
    "FWH the first  hop addition.\r\nAllow last addition to sit for 5 minutes to release aroma.",
  tasteNotes:
    "Very similar to Australian beers in the 60&#39;s.  Pride of Ringwood is the traditional hop used for a very large number of Australian beers. Although not considered a typical flavour hop it works very well as a single hopped beer.  Aim for 50 - 100 ppm of C",
  tasteRating: "41.0000000",
  og: 1.044,
  fg: 1.008,
  carbonation: "2.4000000",
  fermentationStages: "2",
  primaryAge: "4.0000000",
  primaryTemp: "19.4444444",
  secondaryAge: "10.0000000",
  secondaryTemp: "19.4444444",
  tertiaryAge: "7.0000000",
  age: "30.0000000",
  ageTemp: "18.3333333",
  carbonationUsed: "Bottle with 133.96 g Corn Sugar",
  forcedCarbonation: true,
  primingSugarName: "Corn Sugar",
  primingSugarEquiv: "1.0000000",
  kegPrimingFactor: "0.5000000",
  carbonationTemp: "21.1111111",
  displayCarbTemp: "21.1 C",
  date: "14 May 2011",
  estOg: "1.044 SG",
  estFg: "1.008 SG",
  estColor: "16.8 EBC",
  ibu: "28.0 IBUs",
  ibuMethod: "Tinseth",
  estAbv: "4.7 %",
  abv: "4.7 %",
  actualEfficiency: "68.0 %",
  calories: "405.4 kcal/l",
  displayBatchSize: "23.02 l",
  displayBoilSize: "37.12 l",
  displayOg: "1.044 SG",
  displayFg: "1.008 SG",
  displayPrimaryTemp: "19.4 C",
  displaySecondaryTemp: "19.4 C",
  displayTertiaryTemp: "18.3 C",
  displayAgeTemp: "18.3 C"
};
