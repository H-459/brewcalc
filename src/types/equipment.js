// @flow
export type Equipment = {
  batchSize: number,
  boilSize: number,
  efficiency: number,
  evapRate: number,
  coolingLossPct: number,
  trubChillerLoss: number,
  lauterDeadspace: number,
  topUpKettle: number,
  BIAB: boolean
};
