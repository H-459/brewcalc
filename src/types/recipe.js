// @flow
import type { Equipment } from './equipment'
import type { Fermentable } from './fermentable'
import type { Yeast } from './yeast'
import type { Hop } from './hop'
import type { Mash } from './mash'

export type Recipe = {
  // Beer XML 1.0 Required Fields
  //name: string,
  //version: number,
  type: "Extract" | "All Grain" | "Partial Mash",
  //style
  //brewer: string,
  batchSize: number,
  boilSize: number,
  //boilTime: number,
  hops: Array<Hop>,
  fermentables: Array<Fermentable>,
  yeasts: Array<Yeast>,
  //miscs: Array<Misc>,
  //waters: Array<Water>,
  mash: Mash
};

export const RecipeTypes = {
  extract: 'Extract',
  partialMash: 'Partial Mash',
  allGrain: 'All Grain'
}
