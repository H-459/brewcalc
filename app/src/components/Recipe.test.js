import React from "react";
import Recipe from "./Recipe";
import { recipe } from "../data/recipe";
import { equipment } from "../data/equipment";
import renderer from "react-test-renderer";

test("renders without crashing", () => {
  const tree = renderer
    .create(<Recipe recipe={recipe} equipment={equipment} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders without crashing with null equipment", () => {
  const tree = renderer
    .create(<Recipe recipe={recipe} equipment={null} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
