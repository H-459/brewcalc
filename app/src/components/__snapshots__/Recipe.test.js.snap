// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Recipe Specs and Equipment
    </div>
    <div
      className="card-body"
    >
      <div
        className="show-grid row"
      >
        <div
          className="col-md-6"
        >
          <div>
            <div
              title="Name of the recipe"
            >
               
              <b>
                Name
              </b>
               
              Aussie Ale
               
            </div>
            <div
              title="Name of the brewer"
            >
               
              <b>
                Brewer
              </b>
               
              Steve Nicholls
               
            </div>
            <div
              title="May be one of “Extract”, “Partial Mash” or “All Grain”"
            >
               
              <b>
                Type
              </b>
               
              All Grain
               
            </div>
            <div
              title="Target size of the finished batch in liters"
            >
              <b>
                Batch Size
                 
              </b>
              23.02 l
               (6.08 gal)
            </div>
            <div
              title="Starting size for the main boil of the wort in liters"
            >
              <b>
                Boil Size
                 
              </b>
              37.12 l
               (9.81 gal)
            </div>
            <div
              title="The total time to boil the wort in minutes"
            >
              <b>
                Boil Time
                 
              </b>
              90 min
            </div>
            <div
              title="The percent brewhouse efficiency to be used for estimating the starting gravity of the beer"
            >
              <b>
                Efficiency
                 
              </b>
              68 %
            </div>
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div>
            <div
              title="Name of the equipment profile – usually a text description of the brewing setup"
            >
               
              <b>
                Equipment Name
              </b>
               
              Pot (13 Gal/50 L) - BIAB
               
            </div>
            <div
              title="The target volume of the batch at the start of fermentation"
            >
              <b>
                Batch Size
                 
              </b>
              23.02 l
               (6.08 gal)
            </div>
            <div
              title="The pre-boil volume used in this particular instance for this equipment setup.  Note that this may be a calculated"
            >
              <b>
                Boil Size
                 
              </b>
              37.12 l
               (9.81 gal)
            </div>
            <div
              title="The volume of wort lost to evaporation per hour of the boil"
            >
              <b>
                Evap Rate
                 
              </b>
              5.36 l
            </div>
            <div
              title="The amount of wort normally lost during transition from the boiler to the fermentation vessel"
            >
              <b>
                Trub Chiller Loss
                 
              </b>
              3.82 l
               (1.01 gal)
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Ingredients
    </div>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Amount
          </th>
          <th>
            Name
          </th>
          <th>
            Type
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              4.5 kg
               (9.91 lb)
            </div>
          </td>
          <td>
            Pale Malt (2 Row) UK
             (
            3 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.52 kg
               (1.14 lb)
            </div>
          </td>
          <td>
            Munich Malt - 20L
             (
            20 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.2 kg
               (0.44 lb)
            </div>
          </td>
          <td>
            Caramel/Crystal Malt - 20L
             (
            20 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.05 kg
               (0.1 lb)
            </div>
          </td>
          <td>
            Roasted Barley
             (
            300 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.26 kg
               (0.58 lb)
            </div>
          </td>
          <td>
            Cane (Beet) Sugar
             (
            0 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            sugar
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              5.21 g
               (0.18 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            60 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              5.21 g
               (0.18 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            45 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              31.23 g
               (1.1 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            15 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              15.68 g
               (0.55 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            0 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            0.13 ml
          </td>
          <td>
            American Ale
             (Attenuation 
            75 %
            , Form 
            liquid
            )
          </td>
          <td>
            Yeast
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Mash Steps
    </div>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Description
          </th>
          <th>
            Step Temperature
          </th>
          <th>
            Step Time
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Mash In
          </td>
          <td
            style={
              Object {
                "display": "flex",
              }
            }
          >
            Add 
            <div>
              <b>
                 
              </b>
              13.16 l
               (3.48 gal)
            </div>
             of water at 
            <div>
              <b>
                 
              </b>
              67 °C
               (152 °F)
            </div>
          </td>
          <td>
            <div>
              <b>
                 
              </b>
              67 °C
               (152 °F)
            </div>
          </td>
          <td>
            60
             min
          </td>
        </tr>
        <tr>
          <td>
            Mash Out
          </td>
          <td
            style={
              Object {
                "display": "flex",
              }
            }
          >
            Add 
            <div>
              <b>
                 
              </b>
              5.51 l
               (1.45 gal)
            </div>
             of water at 
            <div>
              <b>
                 
              </b>
              76 °C
               (168 °F)
            </div>
          </td>
          <td>
            <div>
              <b>
                 
              </b>
              76 °C
               (168 °F)
            </div>
          </td>
          <td>
            10
             min
          </td>
        </tr>
        <tr>
          <td>
            Sparge
          </td>
          <td
            style={
              Object {
                "display": "flex",
              }
            }
          >
            Fly sparge with 
            <div>
              <b>
                 
              </b>
              10.55 l
               (2.79 gal)
            </div>
             water at 
            <div>
              <b>
                 
              </b>
              76 °C
               (168 °F)
            </div>
          </td>
          <td>
            <div>
              <b>
                 
              </b>
              76 °C
               (168 °F)
            </div>
          </td>
          <td>
            10
             min
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Gravity, Alcohol Content and Color
    </div>
    <div
      className="card-body"
    >
      <div
        className="show-grid row"
      >
        <div
          className="col-md-6"
        >
          <div>
            <div
              title="The estimated original gravity of this recipe"
            >
              <b>
                Original Gravity
                 
              </b>
              1.044 SG
               (10.964 °P)
            </div>
            <div
              title="The estimated final gravity of this recipe"
            >
              <b>
                Final Gravity
                 
              </b>
              1.011 SG
               (2.817 °P)
            </div>
            <div
              title="The bitterness of the recipe as measured in International Bitterness Units"
            >
              <b>
                Bitterness (IBUs): 
              </b>
              28 IBUs
               by Tinseth formula
            </div>
            <div
              title="Estimated color of this beer"
            >
              <b>
                Color
                 
              </b>
              8 SRM
               (16 EBC)
            </div>
            <div
              title="The estimated alcohol by volume for this recipe"
            >
              <b>
                Alcohol by volume : 
              </b>
              4 %
            </div>
            <div
              title="Calories in one liter of the beer, based on original and final gravities"
            >
              <b>
                Calories: 
              </b>
              404
               per one L
            </div>
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div
            style={
              Object {
                "backgroundColor": "rgb(208, 91, 14)",
                "height": 150,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders without crashing with null equipment 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Recipe Specs and Equipment
    </div>
    <div
      className="card-body"
    >
      <div
        className="show-grid row"
      >
        <div
          className="col-md-6"
        >
          <div>
            <div
              title="Name of the recipe"
            >
               
              <b>
                Name
              </b>
               
              Aussie Ale
               
            </div>
            <div
              title="Name of the brewer"
            >
               
              <b>
                Brewer
              </b>
               
              Steve Nicholls
               
            </div>
            <div
              title="May be one of “Extract”, “Partial Mash” or “All Grain”"
            >
               
              <b>
                Type
              </b>
               
              All Grain
               
            </div>
            <div
              title="Target size of the finished batch in liters"
            >
              <b>
                Batch Size
                 
              </b>
              23.02 l
               (6.08 gal)
            </div>
            <div
              title="Starting size for the main boil of the wort in liters"
            >
              <b>
                Boil Size
                 
              </b>
              37.12 l
               (9.81 gal)
            </div>
            <div
              title="The total time to boil the wort in minutes"
            >
              <b>
                Boil Time
                 
              </b>
              90 min
            </div>
            <div
              title="The percent brewhouse efficiency to be used for estimating the starting gravity of the beer"
            >
              <b>
                Efficiency
                 
              </b>
              68 %
            </div>
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div>
            Equipment is not set
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Ingredients
    </div>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Amount
          </th>
          <th>
            Name
          </th>
          <th>
            Type
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              4.5 kg
               (9.91 lb)
            </div>
          </td>
          <td>
            Pale Malt (2 Row) UK
             (
            3 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.52 kg
               (1.14 lb)
            </div>
          </td>
          <td>
            Munich Malt - 20L
             (
            20 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.2 kg
               (0.44 lb)
            </div>
          </td>
          <td>
            Caramel/Crystal Malt - 20L
             (
            20 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.05 kg
               (0.1 lb)
            </div>
          </td>
          <td>
            Roasted Barley
             (
            300 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            grain
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              0.26 kg
               (0.58 lb)
            </div>
          </td>
          <td>
            Cane (Beet) Sugar
             (
            0 SRM
            )
          </td>
          <td
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            sugar
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              5.21 g
               (0.18 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            60 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              5.21 g
               (0.18 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            45 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              31.23 g
               (1.1 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            15 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <b>
                 
              </b>
              15.68 g
               (0.55 oz)
            </div>
          </td>
          <td>
            Pride of Ringwood
             (
            10 %
             Alpha, Boil time
             
            0 min
            )
          </td>
          <td>
            Hop
          </td>
        </tr>
        <tr>
          <td>
            0.13 ml
          </td>
          <td>
            American Ale
             (Attenuation 
            75 %
            , Form 
            liquid
            )
          </td>
          <td>
            Yeast
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Gravity, Alcohol Content and Color
    </div>
    <div
      className="card-body"
    >
      <div
        className="show-grid row"
      >
        <div
          className="col-md-6"
        >
          <div>
            <div
              title="The estimated original gravity of this recipe"
            >
              <b>
                Original Gravity
                 
              </b>
              1.044 SG
               (10.964 °P)
            </div>
            <div
              title="The estimated final gravity of this recipe"
            >
              <b>
                Final Gravity
                 
              </b>
              1.011 SG
               (2.817 °P)
            </div>
            <div
              title="The bitterness of the recipe as measured in International Bitterness Units"
            >
              <b>
                Bitterness (IBUs): 
              </b>
              28 IBUs
               by Tinseth formula
            </div>
            <div
              title="Estimated color of this beer"
            >
              <b>
                Color
                 
              </b>
              8 SRM
               (16 EBC)
            </div>
            <div
              title="The estimated alcohol by volume for this recipe"
            >
              <b>
                Alcohol by volume : 
              </b>
              4 %
            </div>
            <div
              title="Calories in one liter of the beer, based on original and final gravities"
            >
              <b>
                Calories: 
              </b>
              404
               per one L
            </div>
          </div>
        </div>
        <div
          className="col-md-6"
        >
          <div
            style={
              Object {
                "backgroundColor": "rgb(208, 91, 14)",
                "height": 150,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
